src/coreclr/src/jit/gentree.h

```diff
@@ -4501,12 +4586,65 @@ struct GenTreeIntrinsic : public GenTreeOp
 
 struct GenTreeJitIntrinsic : public GenTreeOp
 {
-    var_types gtSIMDBaseType; // SIMD vector base type
-    unsigned  gtSIMDSize;     // SIMD vector size in bytes, use 0 for scalar intrinsics
+private:
+    ClassLayout* m_layout;
+
+    union {
+        var_types      gtOtherBaseType; // For AVX2 Gather* intrinsics
+        regNumberSmall gtOtherReg;      // For intrinsics that return 2 registers
+    };
+
+public:
+    var_types     gtSIMDBaseType; // SIMD vector base type
+    unsigned char gtSIMDSize;     // SIMD vector size in bytes, use 0 for scalar intrinsics
```

Comment:
By consolidating this information on `GenTreeJitIntrinsic` we can pack it better and it allows us to have a `ClassLayout`, which is not only useful for the multireg case, but could both streamline and improve our ability to associate the right class handle with SIMD and HW intrinsics.